python/ray/data/_internal/planner/exchange/sort_task_spec.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Callable, List, Tuple, TypeVar, Union
+
+import numpy as np
+
+from ray.data._internal.progress_bar import ProgressBar
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.planner.exchange.interfaces import ExchangeTaskSpec
+from ray.data.block import Block, BlockAccessor, BlockExecStats, BlockMetadata
+from ray.types import ObjectRef
+
+
+T = TypeVar("T")
+
+# Data can be sorted by value (None), a list of columns and
+# ascending/descending orders (List), or a custom transform function
+# (Callable).
+SortKeyT = Union[None, List[Tuple[str, str]], Callable[[T], Any]]
+
+
+class SortTaskSpec(ExchangeTaskSpec):
+    """
```

Comment:
This file is copied from https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/sort.py, with change from `BlockList` to `RefBundle`.